### ISSUE TITLE
Data won't load in 4.3RC

### ISSUE THREAD

**Issue by Vidyut** [makehumancommunity/mpfb2#218]
**Describe the bug**
I'm testing Blender 4.3 Release Candidate. It is working normally. However, I can't get MPFB to work. The data directory doesn't load.

**Checklist**
- If you are using blender 4, did you use the blender 4 specific build?
I think so. Same thing was working in 4.2
- If you recently upgraded mpfb2, did you read http://static.makehumancommunity.org/mpfb/faq/ensure_correct_version.html
MPFB2 was working in Blender 4.2. Only thing changed is the Blender version.

**Platform and versions**
 - OS: Is this windows, mac or linux?
 Linux. Ubuntu 24.04.1 LTS
 - Version of Blender
 Blender 4.3 Release Candidate
 - Version of MPFB, preferably a link to which exact binary you downloaded. If not possible, an approximate date of when you downloaded MPFB.
Probably downloaded it last week.
mpfb2-20241113.zip
I think the current release is after that, but the github doesn't show changes (as you'd suggested checking) and I'm trying to download it for the few hour got some power issues here and the download is very slow and keeps getting interrupted. Will update if installing the new file solves it when the download completes.

**Comment by joepal1976** [comment]
I tried now, on ubuntu 22.04. I had an installation of blender 4.2.1, downloaded 4.3.0 and made sure to click "import preferences for blender 4.2" when I started it the first time. For me, the data showed up as expected.

I'd need to see how the data directories look to figure out what's wrong. Can you post a screenshot of the preferences window in 4.3.0? And then in mpfb in blender 4.3.0 click "system and resources" -> "directories" -> "user files", open the "data" directory and post a screenshot of that?

If the data directory is empty, can you open the 4.2 user directory and look at the files there? In the file browser you should be able to go to your home directory, click ctrl-h to show hidden files and then navigate to .config/blender/4.2/mpfb/data

If the 42 data directory contains files whereas the 4.3 does not, I'd suggest copying the files to the new location and see if it helps.

**Comment by Vidyut** [comment]
Thanks for these suggestions. I'll do them in a day or two and get back. The path in the preferences is correct (installed and reinstalled so many times, I know the drill...) I'll check all the rest as soon as I get time. This is much appreciated.

**Comment by Vidyut** [comment]
I am now on 4.3. Spent some time on this.

![image](https://github.com/user-attachments/assets/0423cad0-215f-4837-b969-d167d7f18f0a)

Data is present in 4.2. The correct directory is set in MPFB2, but the addon is showing it isn't there. This is the directory.

![image](https://github.com/user-attachments/assets/38781dfd-196d-4cee-b187-5cadf1ccdb45)

The mpfb data folder in 4.2 is empty (I don't recall checking it before, so I don't know if it used to have anything.)
![image](https://github.com/user-attachments/assets/4177531c-adef-4f87-bd6e-f7cce80cf509)

But the one in 4.3 also is empty.
![image](https://github.com/user-attachments/assets/87061e48-2d16-42fc-b303-efa59b9bc884)

Just out of curiosity, I went and checked 4.1 and that's empty too. But MPFB2 was working fine in 4.1 and 4.2, so I don't know. Alternatively, I'll try to reinstall 4.2 and see what happens.

**Comment by Vidyut** [comment]
Reinstalled 4.2. MPFB2 is working normally. Same settings, same files, same everything. Didn't have to change a thing Addon was active when I upgraded, and when I downgraded, it loaded by default with all assets.
![image](https://github.com/user-attachments/assets/47a1d429-df06-421d-a8ec-dfe862065cc3)

**Comment by joepal1976** [comment]
Yes, it seems that the build script had got stuck and stopped pushing new build. I've fixed that now, so the directory should contain the builds again. 

Atm I have no idea what made the assets go missing, so it'd be interesting to see if the same thing happens if you upgrade again. That the data dirs are empty are expected since you explicitly set the "Path to MPFB user data". 

Only thing I can think of is if the path is somehow ended up wrong in the 4.3 settings, such as if you copied it and accidentally got an extra space in the name or something. 

Once you have updated to the latest mpfb nightly build, can you set the log level for locationservice to "DUMP"? 

![image](https://github.com/user-attachments/assets/4a796633-9e02-48fa-8fc4-96e27d96230c)

Then, after having restarted blender, can you click "export log" (in the same panel) with it set to services.locationservice, and attach the file here?

**Comment by joepal1976** [comment]
Also, before restarting, please check the "globally log all uncaught exceptions" box in the addon preferences.

**Comment by Vidyut** [comment]
Will do. Update both MPFB2 and Blender, right. Just confirming.

**Comment by Vidyut** [comment]
Here's what I've done so far
- renamed the 4.3 blender folder so that it reupgrades everything again.
- Upgrading Blender 4.2 > 4.3
- Import Blender 4.2 preferences. Wait... wait... wait... wait...
- Downloading and installing MPFB2 on the fresh install of Blender 4.3 so nothing else can interfere with it.
- Result:
Something strange happened worth mentioning. It already had a path to the data directory. It wasn't the default path you said, nor the path I use, but the directory above it containing the data directory.
![image](https://github.com/user-attachments/assets/919909b5-ccb1-4ec7-b457-b5f5f9269b06)

So I tried to use it as it was and see.
And it worked!
![image](https://github.com/user-attachments/assets/cea026f7-1358-40e4-8a69-08d5d543e46d)

So the change is that the link is no longer to the data directory, but the MPFB directory that contains the data directory (you probably added "data" to the path somewhere, so if we link to data, it gets linked twice... and no data found... I think)

Anyway, this is resolved. If anybody has the same problem, and your user directory url has "data" in it, try removing it to see if that works...

**Comment by Vidyut** [comment]
Um... do you still need the logs or should I disable? In case you need them to track anything.

**Comment by joepal1976** [comment]
No, if it works, all is well.

The mh_user_data stuff is only if you intend to integrate with the MakeHuman standalone application, which is described here https://static.makehumancommunity.org/mpfb/docs/makehuman.html

But if you don't use MakeHuman, then it is not relevant.

**Comment by Vidyut** [comment]
Got it.

BTW, I forgot to disable/uninstall MPFB before upgrading blender so I think this time it was able to read the saved data directory and convert it to the new style automatically. If I had disabled it, the data url goes away and when I plugged in the old one, it didn't work.

(Just in case anybody else runs into trouble with this)
